pahfit/feature_strengths.py

```diff
@@ -103,6 +109,79 @@ def featcombine(ftable):
                          cfdic[feat]['range'][-1],
                          np.sum(dfs[mask]['strength']),
                          None,
-                         None])
+                         np.sum(dfs[mask]['eqw'])
+                         ])
 
     return cftable
+
+
+def eqws(comp_type, x_0, amp, fwhm_stddev, obs_fit):
+    """
+    Calculate the emission features equivalent width
+    (integral[(I_nu-I_cont)/I_cont d_lam]) in microns.
+
+    Parameters
+    ----------
+    comp_type : string
+        type of emission component (Drude1D/Gaussian)
+    x_0 : float
+        central wavelength of the feature.
+    amp : float
+        central intensity of the feature.
+    fwhm_stddev : float
+        fwhm or stddev of the feature depending on comp_type.
+    Returns
+    -------
+    eqw : float
+        the equivalent width of the feature
+    """
+    # Check if the emission component is Gaussian and calculate fwhm.
+    if comp_type == 'Gaussian1D':
+        fwhm = 2 * fwhm_stddev * np.sqrt(2 * np.log(2))
+    else:
+        fwhm = fwhm_stddev
+
+    # Get range and wavelength region for integration.
+    low = x_0 - (fwhm * 6)
+    lmin = low if low > 0 else 0.
+    lmax = x_0 + (fwhm * 6)
+    lam = np.arange(100) / 99 * (lmax - lmin) + lmin
+
+    # Calculate the continuum and feature components in the integration range.
+    cont_components = []
+    for cmodel in obs_fit:
+        if isinstance(cmodel, BlackBody1D):
+            cont_components.append(cmodel)
+    cont_model = cont_components[0]
+    for cmodel in cont_components[1:]:
+        cont_model += cmodel
+    continuum = np.nan_to_num(cont_model(lam))
+
+    if comp_type == 'Drude1D':
+        drude = Drude1D(amplitude=amp,
+                        x_0=x_0,
+                        fwhm=fwhm)
+        lnu = drude(lam)
+
+    elif comp_type == 'Gaussian1D':
+        gauss = Gaussian1D(amplitude=amp,
+                           mean=x_0,
+                           stddev=fwhm_stddev)
+        lnu = gauss(lam)
+
+    # Set broad limit (bl) to replace the continuum in the EQW calculation
```

Comment:
Perhaps add:
# Following the EQW calculation in IDL PAHFIT, we define a default broad limit bl.